GoogleForCreators/web-stories-wp
# UI Simplification: Add Saved Styles into Text Insertion panel

## Summary
Adds Saved Styles section to the Text Insertion panel which allow either inserting a new styled text element or applying style on the text, if selected.


## Relevant Technical Choices



## To-do

- [x] Use the feature flag
- [x] Make style match Figma
- [x] Verify the panel only re-renders when relevant



## User-facing changes

When text is selected (apply mode):
<img width="634" alt="Screenshot 2022-02-24 at 14 56 04" src="https://user-images.githubusercontent.com/3294597/155528122-abe34816-84bc-4666-b829-4ccef46d2a16.png">

When text is not selected (insert mode):
<img width="635" alt="Screenshot 2022-02-24 at 14 56 18" src="https://user-images.githubusercontent.com/3294597/155528185-68e57189-0a17-4940-b3a9-24cfa6f86264.png">




## Testing Instructions






This PR can be tested by following these steps:

Enable the feature flag for Library: Saved Styles first!

1. Open the Library Text Panel.
2. Add a text, e.g. Title and assign some styles to it -- color, font type, background color, etc.
3. Now, in the Library Saved Styles section, click the `+` icon in the panel header.
4. Verify a new style was added based on the selected Text element.
5. Unselect the text element. 
6. Verify that now the `+` icon is disabled.
7. Hover on the added style, verify that a `+` icon appears.
8. Click on the style button, verify a new text is added with the styles applied.
9. Now add a Label
10. Click on the same style button
11. Verify that instead of adding a new element, the selected Text element gets styled.
12. Test also with multi-selection and also open the Style manager from the "More Styles" link and try the same out there.
13. NB -- all this is from the Left side Text Insertion panel and not the Design panel :)


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10113
